pkg/cmd/gist/view/view.go

```diff
@@ -115,11 +121,21 @@ func viewRun(opts *ViewOptions) error {
 			_, err := fmt.Fprint(opts.IO.Out, "\n")
 			return err
 		}
+
 		return nil
 	}
 
 	if opts.Filename != "" {
 		gistFile, ok := gist.Files[opts.Filename]
+		isBinary, err := shared.DetectedBinaryFileContents(gistFile.Content, opts.IsPhysicalFile)
```

Comment:
This check should only happen _after_ the `!ok` check that is currently below. If the filename wasn't found, there is no valid contents to inspect.